### PR TITLE
Implemented dark light theme in setting and use on home title bar

### PR DESCRIPTION
**Pull request by dinkar1708**
### Issue

- Need light dark mode setting.

### Motivation and Context

- Need light dark mode setting.

### Modified points

- Save dark mode setting in preference.
- Restore setting after app open.
- Change the color of title bar based on theme setting.
- Copy past same color as light in dark theme. Modified single color to show the implementation.

### Description and Screenshots
- demo
https://user-images.githubusercontent.com/14831652/202602994-7c366b90-8b75-47d9-ba6e-b1fcaa00faa9.mov
- on off state - title bar color changed.
Simulator Screen Shot - iPhone Xs Max - 2022-11-18 at 11 29 14
Simulator Screen Shot - iPhone Xs Max - 2022-11-18 at 11 29 19

### Figma or screen image URL
- NA

### Worries or Issues

- `(optional)`

### CODE REVIEW

File: lib/presentation/feature/profile/profile_page.dart [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+import '../../common/app_theme_provider.dart';
+
+class ProfilePage extends ConsumerWidget {
+  const ProfilePage({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    return const Center(child: Text('Profile Page'));
+  Widget build(BuildContext context, WidgetRef ref) {
+    var isDarkTheme = ref.watch(appThemeNotifierProvider);
```
Comment (diarmaidlindsay) [inline lib/presentation/feature/profile/profile_page.dart]: change `var` to `final`

File: lib/presentation/feature/profile/profile_page.dart [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            const Text(
+              'Dark Mode',
+            ),
+            Switch(
+              value: isDarkTheme,
+              onChanged: (value) {
+                isDarkTheme = value;
```
Comment (diarmaidlindsay) [inline lib/presentation/feature/profile/profile_page.dart]: This line is unnecessary.
I feel that the following is easier to read : 
```
onChanged: (isDarkTheme) {
                ref
                    .read(appThemeNotifierProvider.notifier)
                    .setIsDarkTheme(isDarkTheme);
              },
```

**Review by diarmaidlindsay — CHANGES_REQUESTED** [review]

**Review by diarmaidlindsay — APPROVED** [review]
LGTM

**Review by enricmacias — APPROVED** [review]
LGTM

**Pull request merged**